go/tools/builders/nogo_change_test.go

```diff
@@ -0,0 +1,786 @@
+package main
+
+import (
+	"fmt"
+	"go/token"
+	"os"
+	"path/filepath"
+	"reflect"
+	"slices"
+	"sort"
+	"testing"
+
+	"golang.org/x/tools/go/analysis"
+)
+
+const (
+	FileA         = "from"
+	FileB         = "to"
+	UnifiedPrefix = "--- " + FileA + "\n+++ " + FileB + "\n"
+)
+
+// Mock helper to create a mock file in the token.FileSet
+func mockFileSet(fileName string, size int) *token.FileSet {
+	fset := token.NewFileSet()
+	f := fset.AddFile(fileName, fset.Base(), size)
+	for i := 0; i < size; i++ {
+		f.AddLine(i)
+	}
+	return fset
+}
+
+// Mock analyzers for the test
+var (
+	analyzer1 = &analysis.Analyzer{Name: "analyzer1"}
+	analyzer2 = &analysis.Analyzer{Name: "analyzer2"}
+)
+
+// ApplyEdits() and validate() here provide the reference implementation for testing
+// applyEditsBytes() from the refactored nogo_change code, now using NogoEdit.
```

Comment:
Please don't do this. If this test fail, people have to debug BOTH your test code AND your production code to see which one has a bug.

Instead, please manually prepare an expected output to compare with the production code

Reply:
done.